### PR TITLE
More code clean-ups, unneeded imports removal, improved input checks.

### PR DESCRIPTION
**Pull request by nbishdev**

### CODE REVIEW

File: tests/test_benchmark.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 import unittest
 
-import numpy as np
 from sparselsh import LSH
-from scipy.sparse import csr_matrix, issparse
+from scipy.sparse import csr_matrix
+import numpy as np
+
+import os
+import timeit
```
Comment (brandonrobertz) [inline tests/test_benchmark.py:8]: The imports here should follow the previous ordering, here they're the opposite of how they should be. https://peps.python.org/pep-0008/#imports

```
Imports should be grouped in the following order:

    Standard library imports.
    Related third party imports.
    Local application/library specific imports.
```

**Review by brandonrobertz — CHANGES_REQUESTED** [review]

File: tests/test_benchmark.py (hunk 1 above)
Comment (nbishdev) [reply tests/test_benchmark.py]: Done. Thanks for the guidelines.

File: sparselsh/lsh.py [hunk 2]
```diff
@@ -326,12 +337,15 @@ def query(self, query_points, num_results=None, distance_func=None,
             raise ValueError(
                 "The distance function %s is invalid." % distance_func
             )
-        if dist_threshold is not None and \
-            (dist_threshold <= 0 or (distance_func == "cosine" and dist_threshold > 1.0)):
+        if dist_threshold and (not isinstance(dist_threshold, int) and \
+            not isinstance(dist_threshold, float)) and \
+            (dist_threshold <= 0 or \
+                (distance_func == "cosine" and dist_threshold > 1.0)):
```
Comment (brandonrobertz) [inline sparselsh/lsh.py:343]: Leave this as it was, please.

File: sparselsh/lsh.py [hunk 3]
```diff
@@ -326,12 +337,15 @@ def query(self, query_points, num_results=None, distance_func=None,
...
             raise ValueError(
                 "The distance threshold %s is invalid." % dist_threshold
             )
-        if num_results is not None and num_results < 1:
+        if num_results and not isinstance(num_results, int) and \
+            num_results < 1:
```
Comment (brandonrobertz) [inline sparselsh/lsh.py:348]: Revert. We don't need to check for int.

File: sparselsh/lsh.py [hunk 4]
```diff
@@ -209,12 +227,12 @@ def index(self, input_points, extra_data=None):
...
-            assert (extra_data is None) or \
-                (extra_data is not None and
-                 input_points.shape[0] == len(extra_data)), \
-                "input_points dimension needs to match extra data dimension"
+        assert issparse(input_points), "input_points needs to be a sparse matrix"
+        assert input_points.shape[1] == self.input_dim, "input_points wrong 2nd dimension"
+        assert input_points.shape[0] == 1 or (input_points.shape[0] > 1 and \
+               (extra_data is None or (isinstance(extra_data, list) and \
+               len(extra_data) == input_points.shape[0]))), \
+               "input_points dimension needs to match extra data dimension"
```
Comment (brandonrobertz) [inline sparselsh/lsh.py:235]: Revert this please. It's unreadable and unnecessarily restricts the input types.

File: sparselsh/lsh.py [hunk 5]
```diff
@@ -58,6 +61,13 @@ class LSH(object):
     def __init__(self, hash_size, input_dim, num_hashtables=1,
                  storage_config=None, matrices_filename=None, overwrite=False):
 
+        assert isinstance(hash_size, int) and hash_size > 0, \
+            "hash_size must be a positive integer"
+        assert isinstance(input_dim, int) and input_dim > 0, \
+            "input_dim must be a positive integer"
+        assert isinstance(num_hashtables, int) and num_hashtables > 0, \
+            "num_hashtables must be a positive integer"
+
```
Comment (brandonrobertz) [inline sparselsh/lsh.py:70]: Remove checks for int type. You can leave value checks.

**Review by brandonrobertz — CHANGES_REQUESTED** [review]

File: sparselsh/lsh.py (hunk 2 above)
Comment (nbishdev) [reply sparselsh/lsh.py]: Fixed.

File: sparselsh/lsh.py (hunk 3 above)
Comment (nbishdev) [reply sparselsh/lsh.py]: Fixed by converting this to an assertion check at the beginning of the function, which is much simpler and easier to read.

File: sparselsh/lsh.py (hunk 5 above)
Comment (nbishdev) [reply sparselsh/lsh.py]: Fixed.

**Comment by nbishdev** [comment]
Ping, since more than 2 months have passed.

**Comment by brandonrobertz** [comment]
Thanks for your PR

**Pull request merged**